### ISSUE TITLE
FlinkCluster sidecar / BeamJob container  : mapped drive handling for input vs output parameters 

### ISSUE THREAD

**Issue by vp999** [GoogleCloudPlatform/flink-on-k8s-operator#304]
I am trying to use common flinkcluster and running different types of job using same cluster (one job at a time)
Here is the setup
1)   FlinkCluster yaml 
![image](https://user-images.githubusercontent.com/31272095/90225276-ae28e900-de2e-11ea-91db-066f39b6a554.png)
   **Data is mapped to etlresearchfileshare**

2) job,yaml
![image](https://user-images.githubusercontent.com/31272095/90225385-e7f9ef80-de2e-11ea-9e90-5c349c3b2458.png)
**Data is mapped to etlresearchfileshare/demo2**
 
parameters : input and output 
![image](https://user-images.githubusercontent.com/31272095/90225496-1b3c7e80-de2f-11ea-9160-d9c1654659f8.png)

Issue :  I have observed that input parameter is job is correctly evalauted to etlresearchfileshare/demo2/input, but output parameters are evaluated as per mounted volume in flinkcluster i.e. evaluated as etlresearchfileshare/output. I could see output file getting created in etlresearchfileshare/output instead of etlresearchfileshare/demo2/output
![image](https://user-images.githubusercontent.com/31272095/90225769-84bc8d00-de2f-11ea-89c3-83ab0e88f24d.png)

**seems discrepancy in handling input vs output parameters .** 
please note that there is no etlresearchfileshare/input folder

**Comment by functicons** [comment]
Yep, the `--input` and `--output` parameters are interpreted by the Beam WordCount job, it is transparent to the operator. Usually in prod environment, you want to use remote storage for both (e.g., HDFS, GCS, S3, Azure Blob Storage, etc).

**Comment by vp999** [comment]
I used here azure file storage.
The problem is different. There are two containers one for side and other for job. Each container has drive mounted on it. I observed that for output parameter, sidecar container drive is used and for input parameter job container drive is used. Kind of mismatch in handling .

**Comment by functicons** [comment]
IIIC, Azure file storage is not Azure blob storage, the former is local file system while the later is distributed file system. Usually we want to use distributed file systems in prod
